src/codelabs/iterables/index.md

```diff
@@ -0,0 +1,571 @@
+These are some of the most common collection types:
+
+* [List][list class]: Used to access elements by their index.
+* [Set][set class]: Used to contain elements that can only occur once.
+* [Map][map class]: Used to access elements using a key.
+
+In Dart, both `List` and `Set` are `Iterable`, meaning that they share the same methods as the 
+`Iterable` class. 
+
+## What is an Iterable?
+
+An `Iterable` is a collection of elements that can be accessed sequentially.
+
+`Iterable`, in Dart, is an abstract class, meaning that you cannot instantiate it directly.
+However you can create a new `Iterable` by creating a new `List` or a `Set`.
+ 
+This example shows a `List` of `int`, which is also an `Iterable` of `int`:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+{% endprettify %}
+
+The difference with a `List`, is that with the `Iterable` you cannot 
+access elements using their index.
+For example **the following code will give you an error**:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable[![1]!];
+{% endprettify %}
+
+If you access elements by the index, the compiler will tell you that
+`The operator '[]' isn't defined for the class 'Iterable'` which
+means that you cannot use `[index]` in this case.
+
+Continue to the next section to learn how to access elements of an `Iterable`.
+
+## Accessing elements
+
+The elements of an `Iterable` can be accessed sequentially.
+
+One way to do that, is by getting an `Iterator` using the `iterator` getter,
+and using it to step through the values. Stepping with the `Iterator` is done
+by calling `Iterator.moveNext`, and if the call returns true, the `Iterator` 
+has now moved to the next element, which is then available as `Iterator.current`. 
+If the call returns `false`, there are no more elements, and `iterator.current` returns `null`.
+
+However, it is easier to use the `for-in loop`, which internally uses an `Iterator` too.
+
+### Example: Using Iterator and for-in loop
+
+The following example shows how to access elements using an `Iterator`
+and a `for-in loop`. These two code snippets give the same results.
+
+[//]: https://gist.github.com/597f152c1df414fc4d7d78d1be22ac22
+<iframe
+  src="{{site.dartpad-embed}}?id=597f152c1df414fc4d7d78d1be22ac22"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="420"
+  width="100%" >
+</iframe>
+
+In the example, you can see how accessing elements using `for-in loop`
+is more convenient than using the `Iterator`.
+The `for-in loop` makes your code easier
+to read and understand, and it is less prone to errors.
+
+{{site.alert.secondary}}
+**Key terms:**
+* **Iterable**: the Dart [Iterable][iterable class] class.
+* **Iterator**: used to access elements from an Iterable.
+* **for-in loop**: an easy way to sequentially access elements from an Iterable.
+{{site.alert.end}}
+
+### Example: Using first and last
```

Comment:
Yes, I agree! I wasn't sure since I didn't want to break the pattern of having 3rd level headers that were not Example or Exercise. Let me know if I should do that change.